engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/process/FlinkSource.scala

```diff
@@ -35,24 +41,30 @@ trait FlinkSource[T] extends Source[T] {
   * Typical source with methods specific for Flink, user has only to define Source
   * @tparam T - type of event that is generated by this source. This is needed to handle e.g. syntax suggestions in UI
   */
-trait BasicFlinkSource[T] extends FlinkSource[T] with ExplicitUidInOperatorsSupport {
+trait BasicFlinkSource[T] extends FlinkSource[T] with SourceTestSupport[T] with ExplicitUidInOperatorsSupport {
 
   def flinkSourceFunction: SourceFunction[T]
 
   def timestampAssigner : Option[TimestampWatermarkHandler[T]]
 
-  def timestampAssignerForTest : Option[TimestampWatermarkHandler[T]] = timestampAssigner
+  override def timestampAssignerForTest : Option[TimestampWatermarkHandler[T]] = timestampAssigner
 
-  override def sourceStream(env: StreamExecutionEnvironment, flinkNodeContext: FlinkCustomNodeContext): DataStream[T] = {
+  def ctxTypeInformation: TypeInformation[Context] = implicitly[TypeInformation[Context]]
```

Comment:
I think we should think about possibility to return not generic info, but based on TypingResult of input...

Reply:
Fixed. TypeInformation is calculated with TypeInformationDetection that is provided by flinkNodeContext.